Compute the viscosity as a function of the temperature and the shear stress

AmelieLaurens opened the pull request:
Organize in classes and modules the code :
- create a file viscosity.yaml to put the required values
- create several classes : Deck : get the value in viscosity.yaml
                                           Polymer : stock the values of deck in variables that we will be reuse
                                           Model : contain the equation to predict the viscosity
                                           Graph : draw the graphic and save it in the folder Graphics
- main_viscosity.py : the only file which needs to be run

AmelieLaurens committed:
reorganization of the code in modules and classes

AmelieLaurens committed:
add the radius as a function of x, the axial position

AmelieLaurens committed:
update master branch

bring manually the branch modules_with_a_separate_class_for_graph to the branch master. This changes had already been review.

AmelieLaurens committed:
organize in classes and modules the code to compute the viscosity

AmelieLaurens committed:
Update README.md

AmelieLaurens committed:
Update README.md

AmelieLaurens committed:
Update README.md

AmelieLaurens committed:
Update README.md

AmelieLaurens committed:
update notations

AmelieLaurens committed:
Update README.md

AmelieLaurens committed:
Update README.md

AmelieLaurens committed:
ajout d'un espace

AmelieLaurens committed:
enlever espace

jaklengaigne commented on requirements.txt:
```diff
@@ -0,0 +1,4 @@
+pyyaml
+numpy
+matplotlib
+math
```
It seems that the math package cannot be installed. 
I am not even sure you actually need it, i was able to run the code on fresh conda environment without it. @ilyasst Can you confirm?
![image](https://user-images.githubusercontent.com/67014404/89250261-5db9bc00-d5e2-11ea-8368-9a1db8a97da9.png)

jaklengaigne replied:
@ilyasst bump?

ilyasst replied:
Sorry missed this one, `math` is part of the Python standard library https://docs.python.org/3/library/math.html so it is always included with Python. It is impossible to install.

jaklengaigne commented on rotary_jet_spinning/datas/data.py:
```diff
@@ -0,0 +1,253 @@
+
+    def final_radius_orifice_radius(self, deck, polymer, machine, model):
+        
+        discretisation = int(deck.doc['Discretisation'])
```
Is there a reason why discretisation is not define as a global variable in the init of the class?
You are using it a lot in the different functions, it seems ineffective to have it redefined each time.

jaklengaigne replied:
Same question for nu?
The kinetic velocity is used in every function. Defining it every time seems counterproductive.

ilyasst commented:
Bump @jaklengaigne :)

jaklengaigne commented on rotary_jet_spinning/datas/data.py:
```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Source : A simple model for nanofiber formation by rotary jet-spinning
```
please at the full bibliographic details here.

What do you think of renaming the class to signal that it is only for mellado's model? When we will add new models or model correction we will be able to keep track.
@ilyasst , can you weight in for such a nomenclature? How do you think we should organize the different models and their corresponding codes?

ilyasst replied:
We could include a .bibfile in the repo (it would be easy as we already have the library on Zotero, we just need to export it as a bib file) and use the unique identifiers for each publication in the python code (such as [mellado2011]). It makes it easier to explore the code and "insert citation" within it. If someone wants the details of the paper, they can just open the bibfile in the repo and access them.

jaklengaigne commented on README.md:
```diff
@@ -2,6 +2,8 @@
 
```
I am going to do a PR to adjust the README.md.

jaklengaigne commented on rotary_jet_spinning/datas/data.py:
```diff
@@ -0,0 +1,253 @@
+        
+
+    
+    def final_radius_omega(self, deck, polymer, machine, model):
```
omega is the angular velocity. In the deck.yaml it is named as such. 
I think that for consistencies, the name should be keep the same. If possible replace omega by angular_velocity where applicable. It won't be pretty in some places, but it will maintain consistency and help readability i think.

jaklengaigne commented on viscosity/models/modelling.py:
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+"""
+Reference : 
```
Excellent! Do the same for Mellado's model.

jaklengaigne reviewed (commented):
The code runs without trouble. Just double check for the math library. Graphs render nicely and are quite clear. I believe the code is nearing its final form and we should soon be able to add some experimental data as examples. I have to main comments regarding the documentation of the code and a maybe more high level discussion on how to implement more user defined variable ranges. See below.


**Documentation:**
I think that the code needs a bit more descriptive comments on what it does. I am thinking especially about the different functions defined in data.py, but that is also the case for the other. 
I am not sure exactly how much we should details, but at least a single sentence at the beginning of each function to state its purpose would be necessary.
This is not regarding the README, as mention in comment, I will be submitting a PR soon about that.

**User input control**
I think we may have to discuss how the code is made to compute the different results. There is a mix of input variables and in-code parameter generation. 
Let me try to explain, the user has to input the machine and polymer parameters (in the deck file), these parameters form a set of unique properties (one single viscosity, one single reservoir size, etc). Then in the Data class, the code itself choose to vary each parameter in a predetermined range to show its effect on the final fiber diameter, keeping the rest of the input fixed to the user-values. For now there is no way for the user to change these ranges without messing in the code itself.
How can we implement these range exploration in the deck input file? Should we even do that?
I think we need a call to discuss that.

AmelieLaurens committed:
add a code to compare the model with experimental values

AmelieLaurens committed:
Update comparison.py

AmelieLaurens committed:
Erase modifications to compare the model with exp (another PR after)

AmelieLaurens committed:
remove math from requirements

AmelieLaurens committed:
erase lines for comparison with experiment

AmelieLaurens committed:
define the discretisation in the init of class GraphFeatures

In the commit before, I forgot to mention in the title but I also define the kinematic viscosity in the init of the class Polymer

AmelieLaurens committed:
replace the name omega by angular_viscosity

AmelieLaurens committed:
add a file for the biblio

ilyasst commented:
@jaklengaigne can you take care of the review for the new version of this PR since you worked with Amelie on the last one ?

AmelieLaurens committed:
add comments to describe what the code does

AmelieLaurens committed:
the user now choose the values of the ranges of each polymer and machine parameters

AmelieLaurens committed:
replace the name omega by angular_velocity

AmelieLaurens committed:
erase the viscosity part of the folder processes_modelling

now the code for the viscosity is in AmelieLaurens/Polymer_properties

AmelieLaurens committed:
add reference

AmelieLaurens committed:
add a markdown file to detail where the mellado equations come from

jaklengaigne commented on main.py:
```diff
@@ -0,0 +1,27 @@
+
+graph = PointGraph(organized_data)
+
+# comparison = Comparison(deck, polymer, machine, model)
```
Just to make sure, the comparison has not yet been implemented (or is not part of this PR)?

AmelieLaurens replied:
It is not part of this PR. I am waiting for this one to be closed before doing a PR for the comparison with experiment part.

jaklengaigne commented on rotary_jet_spinning/models/modelling.py:
```diff
@@ -2,24 +2,24 @@
 # -*- coding: utf-8 -*-
 
 from math import *
```
why do you import all math?
if you are using only a few math function you can import only these. e.g. if you use only sqrt() you can do:
`from math import sqrt`

jaklengaigne reviewed (approved):
The code ran without errors in a fresh conda environment. I am going to approve and merge!
Only one real small comment about the math library. You can do it in the next PR. 

I am going to add an issue to improve separately the readme.

jaklengaigne commented on rotary_jet_spinning/models/modelling.py:
```diff
@@ -34,7 +34,8 @@ def critical_rotational_velocity_threshold(self, surface_tension, orifice_radius
         
     def Initial_velocity(self, omega_th, s0):
```
small error in the name. It should not be a capital first later.

AmelieLaurens replied:
I put a capital to the name of the function because I already used initial_velocity as a variable name. Should I change the name of the function by initial_velocity_fct for example or can I keep the capital letter Initial_velocity ?

jaklengaigne replied:
Ok that is a conflict we shoud resolve. Are both of them referring to the same thing? If not, one of them may be not well named.
Is the variable initialized using the function? If so, what about changing the variable name to v_ini

AmelieLaurens replied:
yes there are referring to the same thing, the variable is initialized with the function yes. ok I changed the variable name to init_velocity

AmelieLaurens committed:
specify which math function is used in the importation command

modification of the way to put references

AmelieLaurens committed:
name changes

the variable initial_velocity became init_velocity
the function Initial_velocity became initial_velocity

ilyasst commented:
Should this one be closed or is it still active ?

jaklengaigne commented:
I believe that it is closed since I merged it.
It was a revert action to correct my wrong merging.
